### ISSUE TITLE
Prefixing archive name breaks live reloading

### ISSUE THREAD

**Issue by mbchangellc** [quarkusio/quarkus#25515]
### Describe the bug

If the archive name is prefixed in a Gradle build, as shown below, live reloading no longer works.  Quarkus detects a class has changed and appears to reload based on logging output but the code change isn't applied.  The attached project illustrates this behavior.  Run the project, modify code in Service.execute() and it won't be reflected after live reload occurs.  Comment out the lines mentioned below and live reloading will work.

```
tasks.withType(Jar) {
    archiveBaseName.set("test-${project.name}")
}
```

### Expected behavior

After live reload occurs, code changes are reflected in the running application.

### Actual behavior

Live reload appears to occur, the expected output is logged, including the name of the class that changed.  After live reloading completes, the changes to the class aren't reflected in the running application.

### How to Reproduce?

Using the attached project:

1. Run project1:quarkusDev
2. Hit http://localhost:8080/test
3. Note the output printed
4. Modify Service.execute() to print different text
5. Wait for live reload to occur
6. Hit http://localhost:8080/test
7. The original text is printed, code change not reflected
[quarkus-gradle-live-reload-bug.zip](https://github.com/quarkusio/quarkus/files/8672554/quarkus-gradle-live-reload-bug.zip)

### Output of `uname -a` or `ver`

Darwin deimos 21.4.0 Darwin Kernel Version 21.4.0: Fri Mar 18 00:45:05 PDT 2022; root:xnu-8020.101.4~15/RELEASE_X86_64 x86_64

### Output of `java -version`

java version "17.0.3" 2022-04-19 LTS Java(TM) SE Runtime Environment (build 17.0.3+8-LTS-111) Java HotSpot(TM) 64-Bit Server VM (build 17.0.3+8-LTS-111, mixed mode, sharing)

### Quarkus version or git rev

2.9.0.Final, 2.8.x.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Gradle 7.4

**Comment by glefloch** [comment]
Thanks for reporting this @mbchangellc, I will have a look at this.

**Comment by glefloch** [comment]
yes it should be, I just don't have the time to do it right now :/
